### ISSUE TITLE
Consider updating Dedup definition to apply de-duplication to error values

### ISSUE THREAD

**Issue by kasei** [w3c/sparql-query#126]
As discussed in #125, `Dedup`/`Distinct` currently does not handle the de-deduplication of error values which is important for handling `COUNT(DISTINCT *)`. This case could be special-cased in the text, but I'd like to see Dedup updated to handle error values natively as it allows custom aggregates to work over error values while having (what I think of as the expected) `DISTINCT` handling handled before the aggregation.

**Comment by hartig** [comment]
As I just wrote in https://github.com/w3c/sparql-query/pull/125#issuecomment-1752146975, the question of error values is irrelevant for `COUNT(DISTINCT *)` (and also for `COUNT(*)`) because these are special cases for which the ListEval function is not even used. Instead, the expected result for these two special cases is defined purely based on the given sequence Ψ of solution mappings, rather than on the sequence M(Ψ) of lists as produced by ListEval.

**Comment by kasei** [comment]
You're right, `ListEval` isn't used in the case of `COUNT(DISTINCT *)`. Including it in the discussion here is a mistake. I think the example you gave in #125 is a good summary of the issue using some arbitrary `SomeSetFct` set function.

**Comment by pfps** [comment]
Is this issue then resolved by #125 being merged?

**Comment by hartig** [comment]
Yes, I would say the issue is resolved. @kasei ?
